backend/src/helpers/organization.ts

```diff
@@ -133,24 +138,76 @@ const createOrganization = async ({
 				description: name
 			});
 
-			organization = await new Organization({
-				name,
-				customerId: customer.id
-			}).save();
-		} else {
-			organization = await new Organization({
-				name
-			}).save();
-		}
+            organization = await new Organization({
+                name,
+                customerId: customer.id
+            }).save();
+        } else {
+            organization = await new Organization({
+                name
+            }).save();
+        }
 
-		await initSubscriptionOrg({ organizationId: organization._id });
-	} catch (err) {
-		Sentry.setUser({ email });
-		Sentry.captureException(err);
-		throw new Error(`Failed to create organization [err=${err}]`);
-	}
+        await initSubscriptionOrg({ organizationId: organization._id });
+    } catch (err) {
+        Sentry.setUser({ email });
+        Sentry.captureException(err);
+        throw new Error(`Failed to create organization [err=${err}]`);
+    }
+
+    return organization;
+};
+
+const deleteOrganization = async ({
+    email,
+    orgId
+}: {
+    email: string;
+    orgId: string;
+}) => {
+    let organization;
+    try {
+        // delete the organization itself
+        const organization = await Organization.findByIdAndDelete(orgId);
 
-	return organization;
+        // delete all the membersOrg
+        await MembershipOrg.deleteMany({
+            organization: orgId
+        });
+
+        // delete all the workspaces
+        // first get all the workspaces ids
+        const workspaceIds = (
+            await Workspace.find({
+                organization: orgId
+            })
+        ).map((m) => m.id);
+
+        // delete all the workspaces one by one
+        for (const id of workspaceIds) {
+            await deleteWorkspace({ id });
```

Comment:
There's been a lot of changes in Infisical since writing this function `deleteWorkspace`; we've added more resources for workspaces.

Can you update this function to also delete:

- All `BotKey`'s with the `workspace`
- All `Folder`'s with the `workspace`
- All `Integration`'s with the `workspace`
- All `IntegrationAuth`'s with the `workspace`
- All `SecretBlindIndexData` with the `workspace`
- All `ServiceAccountKey` with the `workspace`
- All `ServiceAccountWorkspacePermission` with the `workspace`
- All `ServiceToken` with the `workspace`
- All `ServiceTokenData` with the `workspace`
- All `Tag` with the `workspace`

Reply:
sure, i will update the function ASAP 🚀